servicetalk-tcp-netty-internal/src/main/java/io/servicetalk/tcp/netty/internal/TcpConnector.java

```diff
@@ -244,15 +244,16 @@ public void onSubscribe(final Cancellable cancellable) {
                         }
 
                         @Override
-                        public void onSuccess(final C connection) {
-                            requireNonNull(connection);
+                        public void onSuccess(@Nullable final C connection) {
                             if (terminatedUpdater.compareAndSet(ConnectHandler.this, 0, 1)) {
                                 target.onSuccess(connection);
```

Comment:
Do we assume that downstream target `Subscriber` will handle `null` instead of a connection? I think it's easy to assume from the caller side that it's expected to receive non-null connection. Was suggesting to invoke `target.onError` instead of throwing from `onSuccess`.

Reply:
>> Do we assume that downstream target Subscriber will handle null instead of a connection?
Yes, in general in operators we pass-through what we get unless we use it.